debugtools/DDR_VM/src/com/ibm/j9ddr/tools/ddrinteractive/CacheMemory.java

```diff
@@ -0,0 +1,173 @@
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package com.ibm.j9ddr.tools.ddrinteractive;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.nio.ByteOrder;
+import java.util.Properties;
+
+import com.ibm.j9ddr.corereaders.memory.AbstractMemory;
+import com.ibm.j9ddr.corereaders.memory.IAddressSpace;
+import com.ibm.j9ddr.corereaders.Platform;
+import com.ibm.j9ddr.corereaders.memory.IModule;
+import com.ibm.j9ddr.corereaders.memory.IProcess;
+import com.ibm.j9ddr.corereaders.memory.IMemorySource;
+import com.ibm.j9ddr.corereaders.memory.MemoryFault;
+import com.ibm.j9ddr.CorruptDataException;
+import com.ibm.j9ddr.DataUnavailableException;
+import com.ibm.j9ddr.corereaders.osthread.IOSThread;
+
+/**
+ * Object representing a single live
+ * process model where a section of
+ * memory in a the process can be represented
+ * in a byte buffer as its source
+ * @see com.ibm.j9ddr.tools.ddrinteractive.CacheMemorySource
+ *
+ * Can be used for inspection of the
+ * scc, without the need of a core file
+ * For example in the following, it is
+ * used to inspect romMethods:
+ *
+ * <code>
+ * CacheMemory memory = new CacheMemory(ByteOrder.nativeOrder());
+ * IVMData aVMData = VMDataFactory.getVMData((IProcess)memory);
+ * //this address denotes the beginning of range of interest, not shown here
+ * memory.addMemorySource(new CacheMemorySource(sourceStartAddress, size));
+ * aVMData.bootstrap("com.ibm.j9ddr.vm29.SomeDebugHandler");
+ *
+ * public class SomeDebugHandler {
+ *   //can fetch this address from a ROMClass Cookie
+ *   J9ROMClassPointer pointer = J9ROMClassPointer.cast(romStartAddress);
+ *   J9ROMMethodPointer romMethod = pointer.romMethods();
+ *   long dumpFlags = (ByteOrder.nativeOrder() == ByteOrder.BIG_ENDIAN) ? 1 : 0;
+ *   J9BCUtil.j9bcutil_dumpRomMethod(System.out, romMethod, pointer, dumpFlags, J9BCUtil.BCUtil_DumpAnnotations);
+ * }
+ * </code>
+ *
+ * Behaviour of the model is not defined
+ * for the case where the underlying source's
+ * ByteBuffer is modified (address or capacity)
+ * during use.
+ *
+ * @author knewbury01
+ *
+ */
+
+public class CacheMemory extends AbstractMemory implements IProcess
```

Comment:
I think a more general name for this class is appropriate: it can be used for any purpose, not just for access to a shared class cache. I suggest `BufferedMemory` or something similar.

I don't think the placeholder methods (e.g. `getPlatform()` and `getAddressSpace()`) should just return null. Other places in the code base expect those methods to return non-null.